### PR TITLE
defaults: enable 'hlsearch' by default

### PR DESCRIPTION
**Pull request by fmoralesc**
Also, update the documentation regarding the option.

Re: https://github.com/neovim/neovim/issues/2676

### CODE REVIEW

File: test/functional/ui/searchhl_spec.lua [hunk 1]
```diff
@@ -25,20 +25,6 @@ describe('search highlighting', function()
...
-    screen:expect([[
-      some ^text                               |
-      more text                               |
-      ~                                       |
-      ~                                       |
-      ~                                       |
-      ~                                       |
-      /text                                   |
-    ]])
-  end)
```
Comment (justinmk) [inline test/functional/ui/searchhl_spec.lua]: It seems that `set hlsearch` is not tested below, so why not add `execute('set hlsearch')` to line 29 and keep this spec?

File: test/functional/ui/searchhl_spec.lua [hunk 2]
```diff
@@ -25,20 +25,6 @@ describe('search highlighting', function()
...
-
   it('works', function()
     execute('set hlsearch')
```
Comment (justinmk) [inline test/functional/ui/searchhl_spec.lua]: can remove this line

File: test/functional/ui/searchhl_spec.lua (hunk 1 above)
Comment (fmoralesc) [inline test/functional/ui/searchhl_spec.lua]: I made something slightly different, I turned this spec into a test for `:set nohlsearch`.

**Comment by justinmk** [comment]
Merged (with some edits)

**Comment by fmoralesc** [comment]
Thanks.

**Pull request closed without merging**